collections/src/main/scala/strawman/collection/Iterator.scala

```diff
@@ -1123,6 +1123,52 @@ trait Iterator[+A] extends IterableOnce[A] { self =>
     b
   }
 
+  /** Copy values produced by this iterator to an array.
+   *  Fills the given array `xs` starting at index `start`.
+   *  Copying will stop once either the end of the current iterator is reached,
+   *  or the end of the array is reached.
+   *
+   *  @param  xs     the array to fill.
+   *  @param  start  the starting index.
+   *  @tparam B      the type of the elements of the array.
+   *
+   *  @note    Reuse: $consumesIterator
+   *
+   *  @usecase def copyToArray(xs: Array[A], start: Int): Unit
+   *
+   *    $willNotTerminateInf
+   */
+  @`inline` final def copyToArray[B >: A](xs: Array[B], start: Int = 0): xs.type = {
+    copyToArray(xs, start, xs.length)
+  }
```

Comment:
Actually what do you think of using a default parameter value for `len` as well instead of having an overload? (like #456)

Reply:
Ah, that acutally points out that i didn't pick the correct default for `len`! It should have been `size`, not `xs.length`. However, we cannot use `size` (or `length`), as this consumes the iterator. So we have to use a different implementation, I'll push it here.

I think we should not use `size` in Iterable either, as we don't know the cost of the operation.